src/editor/Gui.cpp

```diff
@@ -551,133 +637,256 @@ void ImportWad(Bsp* map, Renderer* app)
 	delete tmpWad;
 }
 
+
 void Gui::drawMenuBar() {
 	ImGuiContext& g = *GImGui;
 
 	ImGui::BeginMainMenuBar();
 
 	if (ImGui::BeginMenu("File"))
 	{
-		if (ImGui::MenuItem("Save", NULL)) {
-			Bsp* map = app->getMapContainingCamera()->map;
-			map->update_ent_lump();
-			//map->write("yabma_move.bsp");
-			//map->write("D:/Steam/steamapps/common/Sven Co-op/svencoop_addon/maps/yabma_move.bsp");
-			map->write(map->path);
+		if (ImGui::BeginMenu("Save")) {
+			for (int r = 0; r < app->mapRenderers.size(); r++)
+			{
+				if (!app->mapRenderers[r]->map)
+					continue;
+				if (ImGui::MenuItem((std::string("Map ") + app->mapRenderers[r]->map->name + ".bsp").c_str())) {
+					Bsp* map = app->mapRenderers[r]->map;
+					if (map)
+					{
+						map->update_ent_lump();
+						map->update_lump_pointers();
+						map->write(map->path);
+					}
+				}
+			}
+			ImGui::EndMenu();
+		}

```

Comment:
Let's keep this simple for now and not allow having multiple maps open at the same time. Copy+paste and some other parts of the UI probably need updating too, and this PR is already pretty big.

Reply:
But it needs to save models with needs coordinates. (It impossible without these option)